biz.aQute.bndlib/src/aQute/bnd/osgi/resource/ResourceUtils.java

```diff
@@ -474,4 +474,12 @@ public static String toProvideCapability(Capability cap) throws Exception {
 		return result;
 	}
 
+	public static boolean isFragment(Resource resource) {
+		List<Capability> capabilities = resource.getCapabilities(IdentityNamespace.IDENTITY_NAMESPACE);
+
+		return (capabilities.size() == 1)
+				&& (capabilities.get(0).getAttributes().get(IdentityNamespace.CAPABILITY_TYPE_ATTRIBUTE).equals(
+						IdentityNamespace.TYPE_FRAGMENT));
```

Comment:
Perhaps to avoid NPE, do `IdentityNamespace.TYPE_FRAGMENT.equals(capabilities.get(0).getAttributes().get(IdentityNamespace.CAPABILITY_TYPE_ATTRIBUTE))`